tests/ui/iter_kv_map.rs

```diff
@@ -0,0 +1,56 @@
+// run-rustfix
+
+#![warn(clippy::iter_kv_map)]
+#![allow(clippy::redundant_clone)]
+#![allow(clippy::suspicious_map)]
+
+use std::collections::{BTreeMap, HashMap};
+
+fn main() {
```

Comment:
Could you add a test with a local closure or external function like this:

```rs
fn magic_map(_: u32, val: u32) -> u32 {
	val
}

// ...

fn main() {
	let some_closure = |(key: u32, _val: u32)| key;

	// For boths maps:

	// Don't lint external funciton
	map.iter().map(magic_map).collect::<Vec<_>>();
	// Don't lint external closure
	map.iter().map(some_closure).collect::<Vec<_>>();
}
```

Here it's fine to only check for the `value` or `key` since this shouldn't be linted. I believe the current implementation already deals with this correctly.

---

One more test I could think of was this:

```rs
map.iter().map(|(_, val)| {val}).collect::<Vec<_>>();
```

This will most likely suggest `values().map(|val| {val})` which is totally fine. That will be caught by the identity lint, if the user doesn't realize that they can remove the second map :upside_down_face: